torch/distributed/rpc/_utils.py

```diff
@@ -1,12 +1,16 @@
 from contextlib import contextmanager
 from typing import cast
+import logging
 from . import api
 from . import TensorPipeAgent
 
+logger = logging.getLogger(__name__)
+
 @contextmanager
 def _group_membership_management(store, name, is_join):
     token_key = "RpcGroupManagementToken"
-    my_token = f"Token{name}-{int(is_join)}"
```

Comment:
Apologize if I missed context here. Any reason that we need different tokens for joins and leave? If that case, do we need to worry about issues if join and leave occur concurrently?

Reply:
> Any reason that we need different tokens for joins and leave?

If we imagine the case where the same token is used for both join and leave and we have `worker0` grabbing the token. On its join, `worker0` grabs `RpcGroupManagementToken` and sets it to `worker0`, all other workers will wait on the `worker0` key. After `worker0` is finished joining, it sets `worker0` key to release all the other workers. Now, when `worker0` is leaving, since we do not perform cleanup on the `worker0` key, then no other members in the group will wait for its leave.

> If that case, do we need to worry about issues if join and leave occur concurrently?

Curious, what is a case when join and leave will happen concurrently? Either way I think that since only one worker can hold the `RpcGroupManagementToken` at a time whether it is for join or leave, it will be okay.

Reply:
Oh I see, I misunderstood. I thought `my_token` will be the only token, and I missed that `token_key` is still there. Changes LGTM